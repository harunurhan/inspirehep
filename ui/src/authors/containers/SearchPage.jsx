import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';
import { connect } from 'react-redux';

import AuthorResultItem from '../components/AuthorResultItem';
import PaginationContainer from '../../common/containers/PaginationContainer';
import ResultsContainer from '../../common/containers/ResultsContainer';
import NumberOfResultsContainer from '../../common/containers/NumberOfResultsContainer';
import LoadingOrChildren from '../../common/components/LoadingOrChildren';

class SearchPage extends Component {
  static renderAuthorItem(result) {
    return <AuthorResultItem metadata={result.get('metadata')} />;
  }

  render() {
    const { loading } = this.props;
    return (
      <Row className="mt3" gutter={32} type="flex" justify="center">
        <Col xs={24} lg={16} xl={16} xxl={14}>
          <LoadingOrChildren loading={loading}>
            <Row type="flex">
              <Col>
                <NumberOfResultsContainer />
              </Col>
            </Row>
            <Row>
              <Col span={24}>
                <ResultsContainer renderItem={SearchPage.renderAuthorItem} />
                <PaginationContainer />
              </Col>
            </Row>
          </LoadingOrChildren>
        </Col>
      </Row>
    );
  }
}

SearchPage.propTypes = {
  loading: PropTypes.bool.isRequired,
};

const stateToProps = state => ({
  loading: state.search.get('loading'),
});

export default connect(stateToProps)(SearchPage);
