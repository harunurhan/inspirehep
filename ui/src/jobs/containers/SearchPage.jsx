import React, { Component } from 'react';
import SearchLayout from './SearchLayout';

class SearchPage extends Component {
  render() {
    return <SearchLayout />;
  }
}

export default SearchPage;
