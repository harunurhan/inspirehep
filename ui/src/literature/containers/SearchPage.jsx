import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import SearchLayout from '../../common/layouts/SearchLayout';

import LiteratureItem from '../components/LiteratureItem';
import search from '../../actions/search';

class SearchPage extends Component {
  componentWillMount() {
    this.props.dispatch(search());
  }

  render() {
    return (
      <SearchLayout
        loading={this.props.loading}
        renderResultItem={result => (
          <LiteratureItem metadata={result.get('metadata')} />
        )}
      />
    );
  }
}

SearchPage.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

const stateToProps = state => ({
  loading: state.search.get('loading'),
});
const dispatchToProps = dispatch => ({ dispatch });

export default connect(stateToProps, dispatchToProps)(SearchPage);
