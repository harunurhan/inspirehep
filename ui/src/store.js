import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
/* eslint-disable import/no-extraneous-dependencies */
import { composeWithDevTools } from 'redux-devtools-extension';

import reducers from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import keepPreviousUrlMiddleware from './middlewares/keepPreviousUrl';
import {
  reHydrateRootStateFromStorage,
  createPersistToStorageMiddleware,
} from './middlewares/statePersister';
import searchDispatcherMiddleware from './middlewares/searchDispatcher';
import redirectToErrorPageMiddleware from './middlewares/redirectToErrorPage';
import actionTrackerMiddleware from './middlewares/actionTracker';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const reduxRouterMiddleware = routerMiddleware(history);

const reducersToPersist = ['ui', 'user'];

const PROD_MIDDLEWARES = [
  reduxRouterMiddleware,
  queryParamsParserMiddleware,
  keepPreviousUrlMiddleware,
  createPersistToStorageMiddleware(reducersToPersist),
  searchDispatcherMiddleware,
  redirectToErrorPageMiddleware,
  actionTrackerMiddleware,
  thunkMiddleware,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(...PROD_MIDDLEWARES);
  }
  return applyMiddleware(...DEV_MIDDLEWARES);
};

export const store = createStore(
  reducers,
  reHydrateRootStateFromStorage(reducersToPersist),
  composeWithDevTools(withMiddlewares())
);
