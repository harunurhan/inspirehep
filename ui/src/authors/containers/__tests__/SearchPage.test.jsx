import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { getStoreWithState } from '../../../fixtures/store';
import SearchPage from '../SearchPage';

describe('SearchPage', () => {
  it('renders with loading state', () => {
    const store = getStoreWithState(
      fromJS({
        search: {
          loading: true,
        },
      })
    );
    const wrapper = shallow(<SearchPage store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
