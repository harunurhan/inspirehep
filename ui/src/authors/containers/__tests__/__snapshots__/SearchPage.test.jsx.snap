// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders with loading state 1`] = `
<Row
  className="mt3"
  gutter={32}
  justify="center"
  type="flex"
>
  <Col
    lg={16}
    xl={16}
    xs={24}
    xxl={14}
  >
    <LoadingOrChildren
      loading={false}
    >
      <Row
        gutter={0}
        type="flex"
      >
        <Col>
          <Connect(NumberOfResults) />
        </Col>
      </Row>
      <Row
        gutter={0}
      >
        <Col
          span={24}
        >
          <Connect(SearchResults)
            renderItem={[Function]}
          />
          <Connect(SearchPagination) />
        </Col>
      </Row>
    </LoadingOrChildren>
  </Col>
</Row>
`;
